xbmc/ServiceBroker.cpp

```diff
@@ -248,9 +248,16 @@ CDatabaseManager& CServiceBroker::GetDatabaseManager()
   return g_application.m_ServiceManager->GetDatabaseManager();
 }
 
-CEventLog& CServiceBroker::GetEventLog()
+CEventLog* CServiceBroker::GetEventLog()
 {
-  return g_serviceBroker.m_pSettingsComponent->GetProfileManager()->GetEventLog();
+  if (g_serviceBroker.m_pSettingsComponent)
+  {
+    auto profileManager = g_serviceBroker.m_pSettingsComponent->GetProfileManager();
+    if (profileManager)
+      return &profileManager->GetEventLog();
+  }
+
```

Comment:
should we log an error here? just in case

Reply:
Not sure we should. Ideally the state of nullptr should be handled (and possibly logged) where called from.
If others want it, i can put in.